packages/flame/lib/src/game/flame_game.dart

```diff
@@ -18,57 +19,90 @@ import 'package:meta/meta.dart';
 ///
 /// This is the recommended base class to use for most games made with Flame.
 /// It is based on the Flame Component System (also known as FCS).
-class FlameGame extends ComponentTreeRoot
+class FlameGame<W extends World> extends ComponentTreeRoot
     with Game
     implements ReadOnlySizeProvider {
   FlameGame({
     super.children,
-    Camera? camera,
-  }) {
+    W? world,
+    CameraComponent? camera,
+    Camera? oldCamera,
+  })  : assert(
+          world is W || (world == null && W == World),
+          'The generics type $W does not conform to the type of '
+          '${world?.runtimeType ?? 'World'}.',
```

Comment:
Wouldn't this be compiler enforced? If you pass an instance of `MyWorld` the `W` is that instance and if you pass a `YourWorld` that does not extends `MyWorld` which is the `W` then it would not allow it to compile right?

Reply:
Yes, that would not allow it to compile, but if you add `MyWorld` as `W` and don't add a world, then the compiler wouldn't complain. The first part of the assertion can be removed though.

Reply:
Yea my comment was more on the first part 😅

Reply:
Makes sense, fixed now :)